### PR TITLE
PMM-6716 Add Show endpoint.

### PR DESCRIPTION
**Pull request by askomorokhov**
[PMM-6716](https://jira.percona.com/browse/PMM-6716)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1177

- https://github.com/percona-platform/dbaas-controller/pull/65
- https://github.com/Percona-Lab/pmm-api-tests/pull/132
- https://github.com/percona/pmm-managed/pull/490
- https://github.com/percona/pmm/pull/594
---
- [x] Tests passed.
- [x] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [x] Update API dependency.
- [x] Update jira ticket description if needed.
- [x] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/dbaas/xtradb_cluster_service.go [hunk 1]
```diff
@@ -86,6 +94,30 @@ func (s XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *dbaas
...
+	// 1. Get pxc host and status
+	//  - Ex.: kubectl get -o=json PerconaXtraDBCluster/<cluster_name>
+	// 2. Get root password:
+	//   - Ex.: kubectl get secret my-cluster-secrets -o json  | jq -r ".data.root" | base64 -d
+	_ = kubernetesCluster
+	resp := dbaasv1beta1.ShowXtraDBClusterResponse{
+		Name:     req.Name,
+		Username: "root",
+		Password: "root_password",
+		Host:     fmt.Sprintf("%s-proxysql", req.Name),
```
Comment (percona-csalguero) [inline services/management/dbaas/xtradb_cluster_service.go:114]: Proxysql?

File: services/management/dbaas/xtradb_cluster_service_test.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ func TestXtraDBClusterService(t *testing.T) {
 		resp, err := s.ListXtraDBClusters(ctx, &dbaasv1beta1.ListXtraDBClustersRequest{KubernetesClusterName: kubernetesClusterNameTest})
 		assert.NoError(t, err)
 		require.NotNil(t, resp.Clusters[0])
-		assert.Equal(t, resp.Clusters[0].Name, "first.pxc.test.percona.com")
+		assert.Equal(t, resp.Clusters[0].Name, "first-pxc-test")
```
Comment (percona-csalguero) [inline services/management/dbaas/xtradb_cluster_service_test.go:136]: Should we also check user and pass are in the response?

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: services/management/dbaas/xtradb_cluster_service.go (hunk 1 above)
Comment (askomorokhov) [reply services/management/dbaas/xtradb_cluster_service.go:114]: looks like yes :)
```
als@mbp pmm-managed % kubectl get -o=json PerconaXtraDBCluster/mycluster | jq -r '.status.host'      
mycluster-proxysql.default
als@mbp pmm-managed % kubectl get secret my-cluster-secrets -o json  | jq -r ".data.root" | base64 -d
root_password%                                                                                                                                                                                                                                                als@mbp pmm-managed % kubectl run -i --rm --tty percona-client --image=percona:8.0 --restart=Never -- bash -il
If you don't see a command prompt, try pressing enter.
bash-4.2$ mysql -hmycluster-proxysql -uroot -proot_password
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 129
Server version: 8.0.18 (ProxySQL)

Copyright (c) 2009-2020 Percona LLC and/or its affiliates
Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show processlist;
+-----+-----------------+---------------------------------------------------------------+--------------------+---------+------+--------------------------+------------------+-----------+---------------+
| Id  | User            | Host                                                          | db                 | Command | Time | State                    | Info             | Rows_sent | Rows_examined |
+-----+-----------------+---------------------------------------------------------------+--------------------+---------+------+--------------------------+------------------+-----------+---------------+
|   1 | system user     |                                                               | NULL               | Sleep   |  858 | innobase_commit_low (-1) | NULL             |         0 |             0 |
|   2 | system user     |                                                               | NULL               | Sleep   |  858 | wsrep aborter idle       | NULL             |         0 |             0 |
|   6 | event_scheduler | localhost                                                     | NULL               | Daemon  |  857 | Waiting on empty queue   | NULL             |         0 |             0 |
|  10 | system user     |                                                               | NULL               | Sleep   |  857 | innobase_commit_low (-1) | NULL             |         0 |             0 |
|  26 | monitor         | 172-17-0-8.mycluster-proxysql.default.svc.cluster.local:51682 | NULL               | Sleep   |    1 |                          | NULL             |         1 |             5 |
| 163 | root            | 172-17-0-8.mycluster-proxysql.default.svc.cluster.local:57444 | information_schema | Query   |    0 | starting                 | show processlist |         0 |             0 |
+-----+-----------------+---------------------------------------------------------------+--------------------+---------+------+--------------------------+------------------+-----------+---------------+
6 rows in set (0.00 sec)

mysql> \s
--------------
mysql  Ver 8.0.20-11 for Linux on x86_64 (Percona Server (GPL), Release 11, Revision 5b5a5d2)

Connection id:          129
Current database:       information_schema
Current user:           root@172-17-0-8.mycluster-proxysql.default.svc.cluster.local
SSL:                    Not in use
Current pager:          stdout
Using outfile:          ''
Using delimiter:        ;
Server version:         8.0.18 (ProxySQL)
Protocol version:       10
Connection:             mycluster-proxysql via TCP/IP
Server characterset:    utf8mb4
Db     characterset:    utf8
Client characterset:    latin1
Conn.  characterset:    latin1
TCP port:               3306
Binary data as:         Hexadecimal
Uptime:                 14 min 46 sec

Threads: 1  Questions: 4  Slow queries: 0
--------------

mysql> 
```

File: services/management/dbaas/xtradb_cluster_service_test.go (hunk 2 above)
Comment (askomorokhov) [reply services/management/dbaas/xtradb_cluster_service_test.go:136]: `ListXtraDBClusters` - should not return creds.

**Review by percona-csalguero — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: services/management/dbaas/xtradb_cluster_service.go [hunk 3]
```diff
@@ -86,6 +94,30 @@ func (s XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *dbaas
...
+
+	// TODO: implement on dbaas-controller side:
+	// 1. Get pxc host and status
+	//  - Ex.: kubectl get -o=json PerconaXtraDBCluster/<cluster_name>
+	// 2. Get root password:
+	//   - Ex.: kubectl get secret my-cluster-secrets -o json  | jq -r ".data.root" | base64 -d
+	_ = kubernetesCluster
+	resp := dbaasv1beta1.ShowXtraDBClusterResponse{
+		Name:     req.Name,
+		Username: "root",
```
Comment (BupycHuk) [inline services/management/dbaas/xtradb_cluster_service.go:112]: have you tested this credentials?
Should they be equal to https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/secrets.yaml?

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Could you add test for ShowXtraDBCluster, please

File: services/management/dbaas/xtradb_cluster_service.go (hunk 3 above)
Comment (askomorokhov) [reply services/management/dbaas/xtradb_cluster_service.go:112]: Yes, I have tested https://github.com/percona/pmm-managed/pull/490/files#r505539895
Thay equal, just base64 decoded.

**Comment by askomorokhov** [comment]
> Could you add test for ShowXtraDBCluster, please

Added

**Comment by askomorokhov** [comment]
> @askomorokhov there is a merge conflict now

Resolved.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**